examples/python/s2ag-datasets.py

```diff
@@ -0,0 +1,60 @@
+"""
+Sample usage of Semantic Scholar Academic Graph Datasets API
+https://api.semanticscholar.org/api-docs/datasets
+"""
+import requests
+import urllib
+import json
+import os
+
+# Get info about the latest release
+latest_release = requests.get("http://api.semanticscholar.org/datasets/v1/release/latest").json()
+print(latest_release['README'])
+print(latest_release['release_id'])
+
+# Get info about past releases
+dataset_ids = requests.get("http://api.semanticscholar.org/datasets/v1/release").json()
+earliest_release = requests.get(f"http://api.semanticscholar.org/datasets/v1/release/{dataset_ids[0]}").json()
+
+# Print names of datasets in the release
+print("\n".join(d['name'] for d in latest_release['datasets']))
+
+# Print README for one of the datasets
+print(latest_release['datasets'][2]['README'])
+
+# Get info about the papers dataset
+papers = requests.get("http://api.semanticscholar.org/datasets/v1/release/latest/dataset/papers",headers={'x-api-key':os.getenv("S2_API_KEY")}).json()
+
+# Download the dataset
+urllib.request.urlretrieve(papers['files'][0], "papers-part0.jsonl.gz")
+
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # 
+# Explore datasets
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+
+## Papers
+papers = [json.loads(l) for l in open("papers/papers-sample.jsonl","r").readlines()]
```

Comment:
Do we need to unzip the downloaded file before executing this line?

Reply:
This script isn't actually runnable. The samples are found elsewhere. How to get them is buried in the README property of a release's description: https://api.semanticscholar.org/datasets/v1/release/latest

reformatted here
```
Semantic Scholar Academic Graph Datasets

These datasets provide a variety of information about research papers taken from a snapshot in time of the Semantic Scholar corpus.

This site is provided by The Allen Institute for Artificial Intelligence (“AI2”) as a service to the
research community. The site is covered by AI2 Terms of Use and Privacy Policy. AI2 does not claim
ownership of any materials on this site unless specifically identified. AI2 does not exercise editorial
control over the contents of this site. AI2 respects the intellectual property rights of others. If
you believe your copyright or trademark is being infringed by something on this site, please follow
the "DMCA Notice" process set out in the Terms of Use (https://allenai.org/terms).

SAMPLE DATA ACCESS
Sample data files can be downloaded with the following UNIX command:

for f in $(curl https://s3-us-west-2.amazonaws.com/ai2-s2ag/samples/MANIFEST.txt)
  do curl --create-dirs "https://s3-us-west-2.amazonaws.com/ai2-s2ag/$f" -o $f
done

See the "s2ag.py" file for Python code that downloads and parses the data.

FULL DATA ACCESS
Downloading the full data requires an API key, which can be obtained at https://www.semanticscholar.org/product/api#Partner-Form
For access to the full datasets, see https://api.semanticscholar.org/api-docs/datasets.

LICENSE and ATTRIBUTION

See the README files for each dataset for information about licensing and attribution.
```